packages/cli/src/commands/build.ts

```diff
@@ -164,51 +186,123 @@ async function run(options: BuildCommandOptions) {
     noDefaults: true,
   });
 
-  const project = new PolywrapProject({
-    rootDir: path.dirname(manifestFile),
-    polywrapManifestPath: manifestFile,
-    logger,
-  });
-  await project.validate();
+  const project = await getProjectFromManifest(manifestFile, logger);
 
-  const polywrapManifest = await project.getManifest();
-  await validateManifestModules(polywrapManifest);
+  if (!project) {
+    return;
+  }
 
-  const buildStrategy = createBuildStrategy(strategy, outputDir, project);
+  await project.validate();
 
-  const schemaComposer = new SchemaComposer({
-    project,
-    client,
-  });
+  const manifest = await project.getManifest();
+  const language = manifest.project.type;
+
+  let execute: () => Promise<boolean>;
 
-  const execute = async (): Promise<boolean> => {
-    const codeGenerator = codegen
-      ? new CodeGenerator({ project, schemaComposer })
-      : undefined;
+  if (isPolywrapManifestLanguage(language)) {
+    await validateManifestModules(manifest as PolywrapManifest);
 
-    const compiler = new Compiler({
-      project,
+    const buildStrategy = createBuildStrategy(
+      strategy,
       outputDir,
-      schemaComposer,
-      buildStrategy,
-      codeGenerator,
-    });
+      project as PolywrapProject
+    );
 
-    const result = await compiler.compile();
+    execute = async (): Promise<boolean> => {
+      const schemaComposer = new SchemaComposer({
+        project,
+        client,
+      });
 
-    if (!result) {
-      return result;
-    }
+      if (codegen) {
+        const codeGenerator = new CodeGenerator({
+          project,
+          schemaComposer,
+          codegenDirAbs: codegenDir,
+        });
+        const codegenSuccess = await codeGenerator.generate();
+
+        if (!codegenSuccess) {
+          logger.error(intlMsg.commands_build_error_codegen_failed());
+          return false;
+        }
+      }
+
+      const compiler = new Compiler({
+        project: project as PolywrapProject,
+        outputDir,
+        schemaComposer,
+        buildStrategy,
+      });
 
-    return true;
-  };
+      const result = await compiler.compile();
+
+      if (!result) {
+        return result;
+      }
+
+      return true;
+    };
+  } else if (isPluginManifestLanguage(language)) {
```

Comment:
I think that we should be handling this inside of the "Compiler", like we already do for interface projects. I'll make a push now to add this.

Reply:
Made these changes here: https://github.com/polywrap/toolchain/pull/1467